### PR TITLE
bpo-33872: Add list elements access complexity

### PR DESCRIPTION
**Pull request by andresdelfino**
https://bugs.python.org/issue33872

### CODE REVIEW

File: Doc/library/stdtypes.rst [hunk 1]
```diff
@@ -1129,7 +1129,7 @@ Lists
 
 Lists are mutable sequences, typically used to store collections of
 homogeneous items (where the precise degree of similarity will vary by
-application).
+application).  List elements are accessed with a O(1).
```
Comment (pablogsal) [inline Doc/library/stdtypes.rst]: Suggestion: "List elements are accessed with O(1) time complexity" or "Accessing list elements has O(1) time complexity".

"List elements are accessed with a O(1)" sounds a bit off to me.

Comment (andresdelfino) [reply Doc/library/stdtypes.rst]: IMHO, mentioning "time complexity" is definitely an improvement, I'll rephrase, thanks :)

File: Doc/library/stdtypes.rst [hunk 2]
```diff
@@ -1131,6 +1131,9 @@ Lists are mutable sequences, typically used to store collections of
 homogeneous items (where the precise degree of similarity will vary by
 application).
 
+Despite its name, lists are more akin to arrays in other languages than to
+linked lists, since its elements are accessed with time complexity O(1).
```
Comment (vadmium) [inline Doc/library/stdtypes.rst]: Use plural: “_their_ name”, “_their_ elements”, to agree with “lists are”

Comment (andresdelfino) [reply Doc/library/stdtypes.rst]: :sweat_smile: thanks!

File: Doc/library/stdtypes.rst [hunk 3]
```diff
@@ -1131,6 +1131,9 @@ Lists are mutable sequences, typically used to store collections of
 homogeneous items (where the precise degree of similarity will vary by
 application).
 
+Despite their name, lists are more akin to arrays in other languages than to
```
Comment (methane) [inline Doc/library/stdtypes.rst:1134]: “despite their name”  seems redundant.
“List” is datastructure independent, abstract word.
For example, C# uses IList and ArrayList for sequence (indexable). And LinkedList for linked list.

**Pull request closed without merging**